### ISSUE TITLE
(Feature request) Move Promoted Attributes to Sidebar

### ISSUE THREAD

**Issue by BeatLink** [TriliumNext/Notes#869]
### Describe feature

With https://github.com/TriliumNext/Notes/pull/538 now implemented, it would be great if all promoted attributes could be moved to the right sidebar to take advantage of horizontal screen real estate

**Comment by eliandoran** [comment]
@BeatLink , I'm not sure if it's a good idea to add them to the sidebar. I'm more inclined toward how Notion handles their equivalent of promoted attributes, which is underneath the title bar.

**Comment by BeatLink** [comment]
What about a toggle to let the user choose?

**Comment by eliandoran** [comment]
@BeatLink , toggles are quite expensive in terms of maintenance. From the moment you add it you now have two more branches in the application. If you make a single change in the promoted attributes (even small changes such as changing a margin), now you have to test then in both scenarios. 

We already added a toggle for the horizontal layout vs vertical layout and it brought a pretty significant costs in terms of maintenance. 

So I would rather pick one and go with it. Maybe adding a toggle/expand button to the existing implementation would also help. 

What would be the use case of having promoted attributes in the sidebar?

**Comment by maphew** [comment]
> What would be the use case of having promoted attributes in the sidebar?

A vertical list is easier to scan through and manage than a horizontal one, especially as the number of attributes grows. Plus as mentioned in the opening, it means more screen real estate is available for the main content when the panel is open.

As for the maintenance burden, is this something that can/should-be offloaded to theme styling?

**Comment by eliandoran** [comment]
@diyoyo , 

> Regarding the latest improvements in the Mobile UI, it's really killing it that all these attributes are now making it impossible to write anything: between the virtual keyboard and the attributes, no space is left for the note itself!

This is not by design, it's actually a bug. On desktop you can scroll if you wish not to see the promoted attributes, on mobile you should be able to do the same. On mobile you should be able to do the same.

In @zadam 's original implementation, how were the promoted attributes displayed on mobile?

**Comment by Shmaur** [comment]
endorse @BeatLink  idea.

Can be provided to the user toggle visible promotion properties.

![Image](https://github.com/user-attachments/assets/4d7d779f-dc2f-4ccd-ae23-95597d0b99bd)

Like the original version.

It is also possible to make the boost appear on the right side of the screen and provide visibility to the user toggle the boost properties.

This will be a very good user interaction.

**Comment by eliandoran** [comment]
@Shmaur , I believe @BeatLink 's idea was to allow the user to select between the original implementation (for which you provided a screenshot) and the existing one, not the collapse feature.

**Comment by Shmaur** [comment]
> [@Shmaur](https://github.com/Shmaur) , I believe [@BeatLink](https://github.com/BeatLink) 's idea was to allow the user to select between the original implementation and sidebar view.
> 
> Collapsing the properties are easier to implement than placing them into two different spots with differing sizes.

I think the collapsed property is better
@eliandoran

**Comment by diyoyo** [comment]
@Shmaur and to answer @eliandoran , I think there was a misunderstanding (probably because we're commenting on this FR by mistake?): I think @Shmaur and I just prefer a non-customizable version where the attributes are presented in column as per latest release, but with the old collapsable button alongside the other "own attributes". Right?

So my question was led by my confusion: I thought you, @eliandoran,  were saying it would be costly to reverse back to "collapsible button of promoted attribute" whereas I now understand you were talking about "making a toggle to switch between two very different layout options".

**Comment by eliandoran** [comment]
@diyoyo ,

> So my question was led by my confusion: I thought you, @eliandoran, were saying it would be costly to reverse back to "collapsible button of promoted attribute" whereas I now understand you were talking about "making a toggle to switch between two very different layout options".

Yes, two different layout options are costly in terms of maintenance. Collapsible promoted attributes is doable, but we need to think of a design for it. A button in the ribbon is not an option since I would prefer to move away from that design pattern.
